lib/pg.utils.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+const RESERVED = require('./sql.reserved');
+
+const escapeString = value => {
+  let backslash = false;
+  let escaped = '\'';
+
+  for (const character of value) {
+    if (character === '\'') {
+      escaped += '\'\'';
+    } else if (character === '\\') {
+      escaped += '\\\\';
+      backslash = true;
+    } else {
+      escaped += character;
+    }
+  }
+
+  escaped += '\'';
+
+  return (backslash ? ` E${escaped}` : escaped);
+};
+
+const PREDEFINED_DOMAINS = {
+  Time: 'time with time zone',
+  DateDay: 'date',
+  DateTime: 'timestamp with time zone',
+  Id: 'bigint',
+  JSON: 'json',
```

Comment:
I don't think it's a good idea, because of:
> When converting textual JSON input into jsonb, the primitive types described by RFC 7159 are effectively mapped onto native PostgreSQL types, as shown in Table 8.23. Therefore, there are some minor additional constraints on what constitutes valid jsonb data that do not apply to the json type, nor to JSON in the abstract, corresponding to limits on what can be represented by the underlying data type. Notably, jsonb will reject numbers that are outside the range of the PostgreSQL numeric data type, while json will not. Such implementation-defined restrictions are permitted by RFC 7159. However, in practice such problems are far more likely to occur in other implementations, as it is common to represent JSON's number primitive type as IEEE 754 double precision floating point (which RFC 7159 explicitly anticipates and allows for). When using JSON as an interchange format with such systems, the danger of losing numeric precision compared to data originally stored by PostgreSQL should be considered.

https://www.postgresql.org/docs/10/static/datatype-json.html

Reply:
Quote from the same page:
> In general, most applications should prefer to store JSON data as `jsonb`, unless there are quite specialized needs, such as legacy assumptions about ordering of object keys.

Also:
> `jsonb` also supports indexing, which can be a significant advantage.